KUBERNETES.md

```diff
@@ -0,0 +1,134 @@
+# Modifying Kubernetes objects
+
+You can delegate Kubernetes object manipulation to the shell-operator.
+
+To do this, you have to write one or more JSON documents describing operation type and its parameters to a file.
+List of possible operations and corresponding JSON specification can be found below.
+
+The path to the file is found in the `$KUBERNETES_OPERATIONS_PATH` environment variable.
+
+> TODO: bash examples
+
+## Operations
+
+### Create
+
+* `operationType` — specifies an operation's type. Deletion types map directly to Kubernetes
+  DELETE's [`propagationPolicy`](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+    * `CreateOrUpdate` — accept a Kubernetes object.
+      It retrieves an object, and if it already exists, computes a JSON Merge Patch and applies it (will not update .status field).
+      If it does not exist, we create the object.
+    * `Create` — will fail if an object already exists
+    * `CreateIfNotExists` — create an object if such an object does not already
+      exist by namespace/name.
+* `object` — full object specification including "apiVersion", "kind" and all necessary metadata.
+
+#### Example
+
+```json
+{
+  "operationType": "CreateOrUpdate",
+  "object": "{\"apiVersion\":\"apps/v1\",\"kind\":\"DaemonSet\",\"metadata\":{\"name\":\"flannel\",\"namespace\":\"d8-flannel\"},\"spec\":{\"selector\":{\"matchLabels\":{\"app\":\"flannel\"}},\"template\":{\"metadata\":{\"labels\":{\"app\":\"flannel\",\"tier\":\"node\"}},\"spec\":{\"containers\":[{\"args\":[\"--ip-masq\",\"--kube-subnet-mgr\"],\"image\":\"flannel:v0.11\",\"name\":\"kube-flannel\",\"securityContext\":{\"privileged\":true}}],\"hostNetwork\":true,\"imagePullSecrets\":[{\"name\":\"registry\"}],\"terminationGracePeriodSeconds\":5}},\"updateStrategy\":{\"type\":\"RollingUpdate\"}}}"
```

Comment:
I think it is better to use object as is, without transforming it into a string. Also, we can support both variants.

Reply:
>Also, we can support both variants.

This seems perfect. I'll change the doc to use a non-ugly object representation. :)